db/collection_policy.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package db
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/core"
+	"github.com/sourcenetwork/defradb/datastore"
+	"github.com/sourcenetwork/defradb/errors"
+
+	ds "github.com/ipfs/go-datastore"
+)
+
+func (db *db) fetchCollectionPolicyDescription(
+	ctx context.Context,
+	txn datastore.Txn,
+	colID uint32,
+) (immutable.Option[client.PolicyDescription], error) {
+	collectionPolicyKey := core.NewCollectionPolicyKey(colID)
+	policyBuf, err := txn.Systemstore().Get(ctx, collectionPolicyKey.ToDS())
+
+	if err != nil && errors.Is(err, ds.ErrNotFound) {
+		return immutable.None[client.PolicyDescription](), nil
+	}
+
+	if err != nil {
+		return immutable.None[client.PolicyDescription](), err
+	}
+
+	var policy client.PolicyDescription
+	err = json.Unmarshal(policyBuf, &policy)
+	if err != nil {
+		return immutable.None[client.PolicyDescription](), err
+	}
+
+	return immutable.Some[client.PolicyDescription](policy), nil
+}
+
+func (c *collection) loadPolicy(ctx context.Context, txn datastore.Txn) error {
```

Comment:
todo: I do not understand why this exists, the policy will have been serialized to json and stored with the rest of the `CollectionDescription` - please remove this function and associated keys.

Reply:
I have to investigate this, might be because the Policy Description stuff was not being persisted, this fixed it. But then later I discovered the serialized JSON was missing the PolicyDescription so I added that which might have had removed the need for this.

- [ ] Remove loadPolicy if not needed

Reply:
> why is the same done for indexes

I cannot answer that :)  Might have been missed in the secondary index review.

Reply:
- [x] Done